### ISSUE TITLE
Keycloak integration breaks on upgrade to >=5.1.1.2

### ISSUE THREAD

**Issue by viktorbard** [WeblateOrg/docker#2048]
### Describe the issue

After upgrading weblate image from 5.1.1.1 to >=5.1.1.2 all users are getting `Could not complete registration. The supplied e-mail address is already in use for another account` when signing in with our keycloak integration.

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar issues in this repository.

### Steps to reproduce the behavior

1. Log in to Weblate with Keycloak as OIDC provider
2. Bump Weblate image from 5.1.1.1 to 5.1.1.2
3. Try signing in to Weblate again with Keycloak

### Expected behavior

Able to map login to existing user, not trying to create a new user.

### Screenshots

Login view:

![image](https://github.com/WeblateOrg/docker/assets/19858249/f6576084-74ca-46a9-aa89-3e2c9d7c430d)

Audit log:

![image](https://github.com/WeblateOrg/docker/assets/19858249/386050bd-169f-4f3f-afcc-a43f71c7a5a9)

**Comment by nijel** [comment]
It upgrades social-auth-core to 4.5.0, see https://github.com/python-social-auth/social-core/releases/tag/4.5.0. The Keycloak change looks harmless: https://github.com/python-social-auth/social-core/pull/815

**Comment by viktorbard** [comment]
Thanks for the links. I agree that they look harmless but after some more investigation we found that default ID-key had changed from `username` to `email` for us.

We tried setting `SOCIAL_AUTH_KEYCLOAK_ID_KEY=username` and `WEBLATE_SOCIAL_AUTH_KEYCLOAK_ID_KEY=username` to override but it didn't work. We ended up changing uid from `username` to `email` in the database then auth worked after upgrade.

Looking at those changes though it looks like default should still be username...
